### PR TITLE
Arvinjayson/td 97 plant information

### PR DESCRIPTION
**Pull request by arvinmybusybee**

### CODE REVIEW

File: src/app/core/api/portal-api.service.ts [hunk 1]
```diff
@@ -243,6 +244,31 @@ export class PortalApi extends BaseApi {
...
         return this.get<IAssetProfile[]>(`${this._apiUrl}/assetprofile/get`);
     }
+
+    getPlantInformations(): Observable<IPlantInformation[]> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation`);
+    }
+
+    getPlantInformation(id: number): Observable<IPlantInformation> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation/${id}`);
+    }
```
Comment (ferlyncaps) [inline src/app/core/api/portal-api.service.ts:254]: we don't have GetPlantInformation  by id. This should be by code as the Plant Information table doesn't have id in the first place

File: src/app/core/api/portal-api.service.ts [hunk 2]
```diff
@@ -243,6 +244,31 @@ export class PortalApi extends BaseApi {
...
+
+    updatePlantInformation(data: any): Observable<any> {
+        return this.put<any>(`${this._apiUrl}/plantinformation`, data);
+    }
+
+    createPlantInformation(data: any): Observable<any> {
+        return this.post<any>(`${this._apiUrl}/plantinformation`, data);
+    }
```
Comment (ferlyncaps) [inline src/app/core/api/portal-api.service.ts:262]: Have you checked the branch I gave? this should be plant-information

File: src/app/core/models/plant-information.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+export interface IPlantInformation {
+    id:number;
```
Comment (ferlyncaps) [inline src/app/core/models/plant-information.ts:2]: where is this came from?

File: src/app/core/services/portal.service.ts [hunk 4]
```diff
@@ -296,37 +297,72 @@ deleteProFormaEntry(id: number): Observable<any> {
...
+    this._portalApi.exportProjectProfiles().subscribe(response => {
+      const contentDisposition = response.headers.get('Content-Disposition');
+      const filename = contentDisposition 
+        ? contentDisposition.split(';')[1].trim().split('=')[1] 
+        : 'Library_Type_Options_' + this._datePipe.transform(new Date(), 'medium') + '.xlsx';
+  
+        const blob = new Blob([response.body], { type: 'application/actet-stream' });
+  
+        const link = document.createElement('a');
+        link.href = window.URL.createObjectURL(blob);
```
Comment (ferlyncaps) [inline src/app/core/services/portal.service.ts:358]: Why are these included in this PR? This is not related in Plant Information

File: src/app/pages/apps/plant-information/plant-information.component.ts [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+      });
+    }
+  }
+
+  onFilterUser(event: any): void {        
+    if (!event.value) {
+      return;
+    }
+    this.types = this.types.filter(t => t.name === event.value);
+  }
```
Comment (ferlyncaps) [inline src/app/pages/apps/plant-information/plant-information.component.ts:321]: Are these lines being used?

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+    this._portalService.exportPlantInformations();
+  }
+
+  onFilterUser(event: any): void {    
+    if (!event.value || event.value == "All") {
+      this.dataSource.data = this.librarytypeoptions;
+      return;
+    }
+    this.dataSource.data = this.librarytypeoptions.filter(t => t.plantCode === event.value);
+  }
```
Comment (ferlyncaps) [inline src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:185]: Are these lines being used?

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+  add() {
```
Comment (ferlyncaps) [inline src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:187]: Please add return type here

File: src/app/pages/apps/plant-information/plant-information.component.html [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+        
+      <div class="flex flex-col flex-auto sm:flex-row sm:gap-4" [style.marginTop.px]="20">
+        <mat-form-field appearance="fill" class="flex-auto">
+            <mat-label>MTD</mat-label>
+            <input matInput formControlName="mtd">
+        </mat-form-field>
+        <!-- <mat-form-field appearance="fill" class="flex-auto">
+            <mat-label>Commisioning Date</mat-label>
+            <input matInput formControlName="commisioningDate" type="date">
+        </mat-form-field> -->
```
Comment (ferlyncaps) [inline src/app/pages/apps/plant-information/plant-information.component.html:85]: please remove these lines

**Review by ferlyncaps — CHANGES_REQUESTED** [review]

File: src/app/core/api/portal-api.service.ts (hunk 1 above)
Comment (arvinmybusybee) [reply src/app/core/api/portal-api.service.ts:254]: Kindly see update

File: src/app/core/api/portal-api.service.ts (hunk 2 above)
Comment (arvinmybusybee) [reply src/app/core/api/portal-api.service.ts:262]: Kindly see update

File: src/app/core/models/plant-information.ts (hunk 3 above)
Comment (arvinmybusybee) [reply src/app/core/models/plant-information.ts:2]: Kindly see update

File: src/app/core/services/portal.service.ts (hunk 4 above)
Comment (arvinmybusybee) [reply src/app/core/services/portal.service.ts:358]: Kindly see update

File: src/app/pages/apps/plant-information/plant-information.component.ts (hunk 5 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/plant-information/plant-information.component.ts:321]: Kindly see update

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts (hunk 6 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:185]: Kindly see update

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts (hunk 7 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:187]: Kindly see update

File: src/app/pages/apps/plant-information/plant-information.component.html (hunk 8 above)
Comment (arvinmybusybee) [reply src/app/pages/apps/plant-information/plant-information.component.html:85]: Kindly see update

File: src/app/core/api/portal-api.service.ts [hunk 9]
```diff
@@ -243,6 +244,31 @@ export class PortalApi extends BaseApi {
...
+
+    getPlantInformations(): Observable<IPlantInformation[]> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation/get`);
+    }
+
+    getPlantInformation(plantCode: string): Observable<IPlantInformation> {
+        return this.get<IPlantInformation>(`${this._apiUrl}/plantinformation/getbycode/${plantCode}`);
+    }
+
+    createPlantInformation(data: any): Observable<any> {
```
Comment (vernienamca) [inline src/app/core/api/portal-api.service.ts:256]: Is it possible to defined the type instead of any?

File: src/app/pages/apps/plant-information/plant-information.component.ts [hunk 10]
```diff
@@ -0,0 +1,324 @@
...
+    });
+
+    const plantCode = this._route.snapshot.paramMap.get('plantcode');
+    this.pageMode = this._route.snapshot.data.pageMode;
+
+    if (this.pageMode === 2) {
+      this._portalService.getPlantInformation(plantCode)
+      .pipe(takeUntil(this._onDestroy$))
+      .subscribe(data => {
+        if (!data) {
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information.component.ts:150]: Fix the code indention.

File: src/app/pages/apps/plant-information/plant-information.component.ts [hunk 11]
```diff
@@ -0,0 +1,324 @@
...
+  createdAt: Date;
+  updatedBy: string;
+  updatedAt: Date;
+  types: ILibraryTypes[];
+  libraryTypes = [];
+  plantCode: string;
+  plantInformationCostCenterData = this.getPlantInformationCostCenterData(0);
+  hasAccess = false;
+
+  costCenterType: any [] = [];
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information.component.ts:39]: Please fix the code indention between code.

File: src/app/pages/apps/plant-information/plant-information.component.ts [hunk 12]
```diff
@@ -0,0 +1,324 @@
...
+ installationType: any [] = [];
+ districtOffice: any [] = [];
+ plantInfoClassification: any [] = [];
+ transmissionGrid: any[] = [];
+ facilityLocation: any[] = [];
+ regions: any[] = [];
+ provinces: any[] = [];
+ municipalities: any[] =[];
+ barangays: any[] =[];
+ costCenterTypes = [{}];//getProducts()
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information.component.ts:50]: Remove this extra code comment.

File: src/app/pages/apps/plant-information/plant-information.component.ts [hunk 13]
```diff
@@ -0,0 +1,324 @@
...
+      costCenterTypeLto: '',
+      action: '',
+    };
+    this.plantInformationCostCenterGrid.collectionView.sourceCollection.unshift(newItem);
+    this.plantInformationCostCenterGrid.collectionView.refresh();
+  }
+
+  onDeleteRow(item: any): void {
+      const index = this.plantInformationCostCenterGrid.collectionView.items.indexOf(item);
+        this.plantInformationCostCenterGrid.collectionView.sourceCollection.splice(index, 1);
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information.component.ts:66]: Fix the code indention.

File: src/app/pages/apps/plant-information/plant-information.component.html [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+</mat-form-field>
+</div>
+
+        
+<hr class="w-full my-8">
+        <div class="table-container" [style.margin-top.px]="50" >
+          <div class="flex flex-row" fxLayoutAlign="end" [style.margin-bottom.px]="10" [style.margin-right.px]="30">
+            <h2 class="m-0 title flex-auto">Cost Center</h2>
+            <button class="rounded-full !bg-green !text-white" mat-raised-button type="button" (click)="addNewRow()">
+              <mat-icon svgIcon="mat:add"></mat-icon>
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information.component.html:182]: Please fix the code indention.

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+    private _router: Router
+  ) {
+  }
+
+  ngOnInit(): void {
+    this._portalService.getPlantInformations()
+    .pipe(
+      takeUntil(this._onDestroy$),
+      finalize(() => this.isListLoading = false))
+    .subscribe(data => {
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:93]: Please fix the code indention.

File: src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+  totalCount: number = 0;
+  pageSize = 10;
+  pageSizeOptions: number[] = [5, 10, 20, 50];
+  dataSource: MatTableDataSource<IPlantInformation> | null;x
+  selection = new SelectionModel<IPlantInformation>(true, []);
+  searchCtrl = new UntypedFormControl();
+  labels = aioTableLabels;      
+  isListLoading = true;  
+  types: ILibraryTypes[];
+  libraryTypes = [];
```
Comment (vernienamca) [inline src/app/pages/apps/plant-information/plant-information-list/plant-information-list.component.ts:70]: Remove the spacing between code.

**Review by vernienamca — CHANGES_REQUESTED** [review]
